docs/proposals/healthchecks.md

```diff
@@ -0,0 +1,67 @@
+| `get`      | string | no       | `exec`            | A URL to do HTTP request on. Detailed information of what field in URL gets converted to what is given below. | `readinessProbe.httpGet` **OR** `livenessProbe.httpGet`                         |
+| `exec`     | string | no       | `get`             | A command to run inside the container.                                                                        | `readinessProbe.exec` **OR** `livenessProbe.exec`                               |
+| `delay`    | uint32 | yes      | None              | Number of seconds after the container has started before probes are initiated.                                | `livenessProbe.initialDelaySeconds` **OR** `readinessProbe.initialDelaySeconds` |
+| `timeout`  | uint32 | yes      | None              | Number of seconds after which the probe times out.                                                            | `livenessProbe.timeoutSeconds` **OR** `readinessProbe.timeoutSeconds`           |
+| `tries`    | uint32 | yes      | None              | Minimum consecutive failures for the probe to be considered failed after, having succeeded.                   | `livenessProbe.failureThreshold` **OR** `readinessProbe.failureThreshold`       |
+
+
+## URL explanation
+
+A URL gets deconstructed as follows
+
+```
+http://localhost:8080/healthz
+```
+
+| Field       | Optional | Details                                                                                                                                                  | Converted to (append to `deployment.spec.template.spec.containers`)                                                            |
+|-------------|----------|----------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| `http`      | Yes      | Scheme to use for connecting to the host.                                                                                                                | `readinessProbe.httpGet.scheme` |
+| `localhost` | No       | If the value is `localhost` or `127.0.0.1` then it is not assigned to anything after conversion. But if the value is some other valid IP it will be set. | `readinessProbe.httpGet.host`   |
+| `8080`      | No       | Number of the port to access on the container. Number must be in the range 1 to 65535.                                                                   | `readinessProbe.httpGet.port`   |
+| `healthz`   | Yes      | Path to access on the HTTP server.                                                                                                                       | `readinessProbe.httpGet.path`   |
+
+**NOTE**: In above converted to field for each URL field if `livenessProbe` is enabled then those respective fields will be created.
+
+## Conflicting fields
```

Comment:
We should think about better design. 
This will involve a lot of verification, it would be better to design data structures so those conflicts can't happen.